### PR TITLE
hide protolude redifined imports

### PR DESCRIPTION
**Pull request by marcosh**
attempt to solve #150

### CODE REVIEW

File: src/GraphQL/Internal/Validation.hs [hunk 1]
```diff
@@ -58,7 +58,7 @@ module GraphQL.Internal.Validation
   , findDuplicates
   ) where
 
-import Protolude hiding ((<>))
+import Protolude hiding ((<>), throwE)
```
Comment (teh) [inline src/GraphQL/Internal/Validation.hs]: Do we have a `throwE` collision as well?

Comment (marcosh) [reply src/GraphQL/Internal/Validation.hs]: it looks like it. After hiding the `Type` from `protolude`, I got the same error with this `throwE`

**Comment by jml** [comment]
Thanks for tackling this, @marcosh 

This change introduces build warnings for users of protolude 0.2.0. I would rather not have the warnings, and think the right thing to do is restrict graphql-api to use protolude 0.2.1 and later.

Concretely, this means:

- [ ] Add a note to the `$next` release section of `CHANGELOG.rst` about the changed protolude requirements
- [ ] Update `package.yaml` to depend on protolude >= 0.2.1
- [ ] Update `stack.yaml` to say `protolude-0.2.1` rather than `protolude-0.2` (not 100% sure this is necessary)

@marcosh, are you OK to make these changes?

Separately, we should do a follow-up PR that breaks the build on warnings.

**Comment by marcosh** [comment]
@jml I updated the PR, hope everything's fine

**Pull request merged**